### ISSUE TITLE
Provide default organization for --org-id in the config file

### ISSUE THREAD

**Issue by canatella** [CircleCI-Public/circleci-cli#832]
## What problem does this feature solve?:
With private orbs, the --org-id argument is mandatory for multiple circleci cli commands, most annoyingly the validate one. Being able to configure a default value in the `~/.circleci/cli.yml` config file would be nice.

## Provide an example:
`~/.circleci/cli.yml`
```
host: https://circleci.com
...
default_org_id: "12345678-1234-1234-1234-123412345678"
```

**Comment by jasonarewhy** [comment]
Agreed, we should be able to configure this in the cli.yml and/or with a `CIRCLECI_CLI_ORG_ID`environment variable.

**Comment by jasonarewhy** [comment]
From [the code](https://github.com/CircleCI-Public/circleci-cli/blob/667c7b3e7e8152ee9bfb37d241eac23dc8d08154/cmd/config.go#L165-L181), it appears `--org-id` is actually optional, so long as you are using `--org-slug`.

We used to be able to use the org slug only but found that recently we had to begin using org-id instead. However, it seems that CircleCI (silently?) renamed org slugs from `github/<org name>` to `gh/<org name>`. So our docs on using `--org-slug` were no longer valid.

When I hit the API `me/collaborations` endpoint to get a list of orgs, I was able to see the _correct_ org slug with the `gh/` prefix.

Hope this helps others frustrated by having to use the non-human-friendly org-id.

**Comment by JulesFaucherre** [comment]
Hello!
@canatella Definitely, thanks for the suggestion, we will consider this and let you know
@jasonarewhy The suggestion for env variable could be interesting, we'll look into this. Regarding your `--org-slug` problem , you're completely right, we introduced this problem by migrating the config validation to a new route. This should be fixed by [this pr](https://github.com/CircleCI-Public/circleci-cli/pull/922) so you should not have the problem anymore. Tell us if you still have it

**Comment by pete-woods** [comment]
The CLI has been rewritten from scratch for v1 (the `main` branch — `v0` is the legacy CLI that ships today and now takes critical fixes only).

**v1 resolves the organization for you**, so there is no mandatory UUID to configure.
Order of resolution for commands that need an org (including `config validate` and
`orb validate`, the annoying ones in this report):

1. `--org` if given — and it takes a slug, `gh/myorg`, not a raw UUID
2. the binding from `circleci project link` for this checkout
3. the git remote

So in a linked checkout, `circleci config validate` with private orbs just works with
no flag and no config-file entry.

Closing as resolved — if you want to pin a default explicitly, `circleci project link`
records it per checkout, which is closer to right than a single global default when you
work across orgs.
